src/Sylius/Bundle/CoreBundle/Doctrine/DQL/Day.php

```diff
@@ -36,12 +38,16 @@ public function parse(Parser $parser): void
 
     public function getSql(SqlWalker $sqlWalker): string
     {
-        $platformName = $sqlWalker->getConnection()->getDatabasePlatform()->getName();
+        $platform = $sqlWalker->getConnection()->getDatabasePlatform();
 
-        return match ($platformName) {
```

Comment:
Better:
```
        return match (get_class($platformName)) {
            MySQLPlatform::class => sprintf('DAY(%s)', $sqlWalker->walkArithmeticPrimary($this->date)),
            PostgreSQLPlatform::class => sprintf('EXTRACT(DAY FROM %s)', $sqlWalker->walkArithmeticPrimary($this->date)),
            default => throw new \RuntimeException(sprintf('Platform "%s" is not supported!', $platformName)),
        };
```
but even that is broken cause `MySQLPlatform::class` was renamed from `MySqlPlatform::class`.

Reply:
I used the `match` expression for the first time but renamed the `MySqlPlatform` class forced me to change the approach to the regular `if`.